ruby-service/app/circuit_breaker.rb

```
@@ -0,0 +1,392 @@
+      end
+    end
+
+    def to_h
+      @mutex.synchronize do
+        {
+          total_calls: @total_calls,
+          successful_calls: @successful_calls,
+          failed_calls: @failed_calls,
+          rejected_calls: @rejected_calls,
+          state_transitions: @state_transitions,
+          average_response_time_ms: (@response_times.empty? ? 0 : @response_times.sum.to_f / @response_times.size * 1000).round(2),
```

RuboCop (1.87.0)

[convention] Line is too long. [131/120]

(Layout/LineLength)